api/v2/performanceprofile_validation_test.go

```diff
@@ -69,6 +86,18 @@ func NewPerformanceProfile(name string) *PerformanceProfile {
 			NUMA: &NUMA{
 				TopologyPolicy: &numaPolicy,
 			},
+			Net: &Net{
+				UserLevelNetworking: pointer.BoolPtr(true),
+				Devices: []Device{
+					{
+						Name:     &netDeviceName,
+						VendorID: &netDeviceVendorID,
+						ModelID:  &netDeviceModelID,
+						PCIpath:  &netDevicePCIpath,
+						MAC:      &netDeviceMAC,
```

Comment:
what about if these fields refer to different cards? should we error out?

Reply:
> /lgtm
> /hold
> All my comments have been addressed, so LGTM, but holding to let other reviewers review.
> I'd add a test (or few tests) to document the behaviour when the user set conflicting parameters to identify a device. What should happen in that case?

I think PPC can check for errors, but PAO can't know if vendorid:deviceid makes sense.
BTW, are we allowed to give both vendor/device AND interfaceName for the the same net device?
Is there any other kind of conflict?

Reply:
yep, close to my point. Do we want to make the ways to identify a device mutually exclusive? do we care at all about possible inconsistencies?

Reply:
I like the "more validation" approach, I would make it mutually exclusive.
Any inconsistencies would result on the tuned profile not being applied partially; we will we end up with a cryptic error event  in the "best-case", silent failure otherwise.

Reply:
Please note that you are commenting on outdate code,  I already removed MAC and PCI PATH.
> BTW, are we allowed to give both vendor/device AND interfaceName for the the same net device?
All the relevant validations are here:
https://github.com/openshift-kni/performance-addon-operators/blob/edac4f260867db0f8edb26ac75f9805b6d757771/api/v2/performanceprofile_validation.go#L270 

Let me know if there is anything else missing there.

Reply:
yes, I seen. The open question still stands: do we want to allow users to specify ALL the parameters or should the ways to identify devices be mutually exclusive? what if interfacename and vendor:device are inconsistent? we just error out?
The answer can totally be "yes, we just error out", but worth discussing anyway.

Reply:
We cant error out on PAO in that case since we cant check in this stage if under the same device object  if interfacename and vendor:device are consistent.  
We can either deny mutual existence of both on PAO  or verify that on PPC side,  not sure what is the best option atm

Reply:
ok, so I guess we keep things as they are atm.

Reply:
> We cant error out on PAO in that case since we cant check in this stage if under the same device object if interfacename and vendor:device are consistent.
> We can either deny mutual existence of both on PAO or verify that on PPC side, not sure what is the best option at the moment.

I vote for denying the use of InterfaceName AND (vendorid-deviceid) pair for the same "netdevice". If you want both, declare 2 netdevices.
PPC will check the semantics of each (not together)

Reply:
Remember the netdevice can be a wildcard. In general my personal experience is to avoid overconstrained systems and checks, it eventually blocks someone from doing something they need.

Reply:
What will be the meaning of NetDevice {interfaceName=ens*,deviceid=0x1000} ? All 0x1000 devices mapped to ens* interfaces? Does tuned support that? Or it will be translates into  0x1000 devices + ens*  interfaces?

Reply:
it will be translated to 0x1000 devices + ens* interfaces

Reply:
So on our system it will look like an "AND" operation, but would be translated into an "OR". Does it look friendly to anyone?